src/python/gudhi/representations/vector_methods.py

```diff
@@ -806,32 +807,49 @@ def fit(self, X, y=None, sample_weight=None):
         Returns:
             self
         """
-        if not hasattr(self.quantiser, 'fit'):
-            raise TypeError("quantiser %s has no `fit` attribute." % (self.quantiser))
-
-        # In fitting we remove infinite death time points so that every center is finite
-        X = [dgm[~np.isinf(dgm).any(axis=1), :] for dgm in X]
+        n_clusters = self.quantiser.n_clusters
 
+        if not len(X):
+            raise Exception("Cannot fit Atol on empty target.")
```

Comment:
Raising a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) would be more appropriate here.